### PR TITLE
Rename "rename-branch" command to "rename"

### PR DESCRIPTION
**Pull request by bengeois**
Fix https://github.com/git-town/git-town/issues/4038

Through this PR, I've tried to rename all occurrences of the `git town rename-branch` command to `git town rename` in the code, documentation and tests

Quite some changes :)
Let me know if you'd like any adjustments!

### CODE REVIEW

File: vendor/github.com/google/go-github/v58/github/repos.go [hunk 1]
```diff
@@ -1440,12 +1440,12 @@ func (s *RepositoriesService) GetBranch(ctx context.Context, owner, repo, branch
 	return b, newResponse(resp), err
 }
 
-// renameBranchRequest represents a request to rename a branch.
```
Comment (kevgo) [inline vendor/github.com/google/go-github/v58/github/repos.go:1443]: Please revert the changes made to this file. This is not part of the Git Town source code, but an external dependency that was downloaded into this monorepo for faster and more reliable builds.

An easy way to revert this is running 

```
git checkout main vendor/github.com/google/go-github/v58/github/repos.go
```

**Review by kevgo — CHANGES_REQUESTED** [review]
Thanks for this very clean and well done change! Just one review comment, then this is good to go!

File: internal/cmd/rename.go [hunk 2]
```diff
@@ -47,17 +47,17 @@ When run on a perennial branch:
...
 - registers the new perennial branch name in the local Git Town configuration`
 
-func renameBranchCommand() *cobra.Command {
+func renameCommand() *cobra.Command {
 	addDryRunFlag, readDryRunFlag := flags.DryRun()
 	addForceFlag, readForceFlag := flags.Force("force rename of perennial branch")
 	addVerboseFlag, readVerboseFlag := flags.Verbose()
 	cmd := cobra.Command{
-		Use:   "rename-branch [<old_branch_name>] <new_branch_name>",
+		Use:   "rename [<old_branch_name>] <new_branch_name>",
```
Comment (stephenwade) [inline internal/cmd/rename.go:55]: @kevgo 2 questions:

1. Should we keep an undocumented `rename-branch` command so this isn't a breaking change?
2. Does Git Town have the ability to update the local Git aliases from `git rename-branch` to `git rename` (or detect old alises and suggest that the user update them)?

Comment (kevgo) [reply internal/cmd/rename.go:55]: Re 1: yes, I'll add that separately
Re 2: That's a great idea, will also implement separately

File: vendor/github.com/google/go-github/v58/github/repos.go (hunk 1 above)
Comment (bengeois) [reply vendor/github.com/google/go-github/v58/github/repos.go]: Yes you're right, missing that wrong changes, I've reverted this change

File: internal/cmd/rename.go (hunk 2 above)
Comment (bengeois) [reply internal/cmd/rename.go:55]: Agree with the second point. I was thinking that day how git-town would handle changes in aliases.

Comment (kevgo) [reply internal/cmd/rename.go:55]: Git Town already automatically updates the configuration settings stored in Git metadata when their names change. We can use this mechanism to update the aliases, which are also stored in Git metadata.

**Review by kevgo — APPROVED** [review]
